### ISSUE TITLE
Not able to build OpenSSL 3.0 for VC-WIN64A if I set the --prefix and --openssldir option in configure

### ISSUE THREAD

**Issue by rakesh-mahapatra** [openssl/openssl#19527]
Not able to build OpenSSL 3.0.5 for VC-WIN64A if I set the --prefix and --openssldir option in configure.
If I don't set these options, then  the build is happening but install won't happen as I don't have admin rights always. I can provide admin credentials, if it asks for it during install.

Below is the command that I used for configure.

1. `perl Configure enable-fips --prefix=C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12 --openssldir=C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12`

**Output:**

```
Configuring OpenSSL version 3.0.5 for target VC-WIN64A
Using os-specific seed configuration
Created configdata.pm
Running configdata.pm
Created makefile.in
Created makefile

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL.md file first)      ***
***                                                                ***
**********************************************************************
```

2. Output of perl configdata.pm --dump

```
Perl information:

    C:\Strawberry\perl\bin\perl.exe
    5.32.1 for MSWin32-x64-multi-thread

Enabled features:

    acvp-tests
    aria
    asm
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    fips
    fips-securitychecks
    gost
    idea
    legacy
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    uplink
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    afalgeng            [not-linux] OPENSSL_NO_AFALGENG
    asan                [default]   OPENSSL_NO_ASAN
    buildtest-c++       [default]
    crypto-mdebug       [default]   OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]   OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]   OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]   OPENSSL_NO_EGD
    external-tests      [default]   OPENSSL_NO_EXTERNAL_TESTS
    fuzz-afl            [default]   OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]   OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [default]   OPENSSL_NO_KTLS
    md2                 [default]   OPENSSL_NO_MD2 (skip crypto\md2)
    msan                [default]   OPENSSL_NO_MSAN
    rc5                 [default]   OPENSSL_NO_RC5 (skip crypto\rc5)
    sctp                [default]   OPENSSL_NO_SCTP
    trace               [default]   OPENSSL_NO_TRACE
    ubsan               [default]   OPENSSL_NO_UBSAN
    unit-test           [default]   OPENSSL_NO_UNIT_TEST
    weak-ssl-ciphers    [default]   OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]
    zlib-dynamic        [default]
    ssl3                [default]   OPENSSL_NO_SSL3
    ssl3-method         [default]   OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "lib",
    ARFLAGS => "/nologo",
    AS => "nasm",
    ASFLAGS => "-g",
    CC => "cl",
    CFLAGS => "/W3 /wd4090 /nologo /O2",
    CPP => "\"\$(CC)\" /EP /C",
    HASHBANGPERL => "/usr/bin/env perl",
    LD => "link",
    LDFLAGS => "/nologo /debug",
    MT => "mt",
    MTFLAGS => "-nologo",
    RANLIB => "CODE(0x2612968)",
    RC => "rc",
    ar_resp_delim => "
",
    aroutflag => "/out:",
    asflags => "-Ox -f win64 -DNEAR",
    asm_arch => "x86_64",
    asoutflag => "-o ",
    bin_cflags => "/Zi /Fdapp.pdb",
    bin_lflags => "setargv.obj /subsystem:console /opt:ref",
    bn_ops => "SIXTY_FOUR_BIT",
    build_file => "makefile",
    build_scheme => [ "unified", "windows", "VC-common" ],
    cflags => "/Gs0 /GF /Gy /MD",
    coutflag => "/Fo",
    cppflags => "",
    defines => [ "OPENSSL_BUILDING_OPENSSL", "OPENSSL_SYS_WIN32", "WIN32_LEAN_AND_MEAN", "UNICODE", "_UNICODE", "_CRT_SECURE_NO_DEPRECATE", "_WINSOCK_DEPRECATED_NO_WARNINGS" ],
    disable => [  ],
    dso_cflags => "/Zi /Fddso.pdb",
    dso_scheme => "win32",
    enable => [  ],
    ex_libs => "ws2_32.lib gdi32.lib advapi32.lib crypt32.lib user32.lib",
    includes => [  ],
    ld_implib_flag => "/implib:",
    ld_resp_delim => "
",
    lddefflag => "/def:",
    ldoutflag => "/out:",
    ldpostoutflag => "",
    ldresflag => " ",
    lflags => "",
    lib_cflags => "/Zi /Fdossl_static.pdb",
    lib_cppflags => "",
    lib_defines => [ "L_ENDIAN" ],
    makedep_scheme => "VC",
    makedepcmd => "\"\$(CC)\" /Zs /showIncludes",
    module_cflags => "",
    module_cxxflags => undef,
    module_ldflags => "/dll",
    mtinflag => "-manifest ",
    mtoutflag => "-outputresource:",
    multilib => "-x64",
    perl_platform => "Windows::MSVC",
    perlasm_scheme => "auto",
    rcoutflag => "/fo",
    shared_cflag => "",
    shared_defflag => "",
    shared_defines => [  ],
    shared_ldflag => "/dll",
    shared_rcflag => "",
    shared_target => "win-shared",
    sys_id => "WIN64A",
    thread_defines => [  ],
    thread_scheme => "winthreads",
    unistd => "<unistd.h>",
    uplink_arch => "x86_64",

Recorded environment:

    AR =
    ARFLAGS =
    AS =
    ASFLAGS =
    BUILDFILE =
    CC =
    CFLAGS =
    CPP =
    CPPDEFINES =
    CPPFLAGS =
    CPPINCLUDES =
    CROSS_COMPILE =
    CXX =
    CXXFLAGS =
    HASHBANGPERL =
    LD =
    LDFLAGS =
    LDLIBS =
    MT =
    MTFLAGS =
    OPENSSL_LOCAL_CONFIG_DIR =
    PERL =
    RANLIB =
    RC =
    RCFLAGS =
    RM =
    WINDRES =
    __CNF_CFLAGS =
    __CNF_CPPDEFINES =
    __CNF_CPPFLAGS =
    __CNF_CPPINCLUDES =
    __CNF_CXXFLAGS =
    __CNF_LDFLAGS =
    __CNF_LDLIBS =

Makevars:

    AR              = lib
    ARFLAGS         = /nologo
    AS              = nasm
    ASFLAGS         = -g
    CC              = cl
    CFLAGS          = /W3 /wd4090 /nologo /O2
    CPP             = "$(CC)" /EP /C
    CPPDEFINES      =
    CPPFLAGS        =
    CPPINCLUDES     =
    CXXFLAGS        =
    HASHBANGPERL    = /usr/bin/env perl
    LD              = link
    LDFLAGS         = /nologo /debug
    LDLIBS          =
    MT              = mt
    MTFLAGS         = -nologo
    PERL            = C:\Strawberry\perl\bin\perl.exe
    RANLIB          = ranlib
    RC              = rc
    RCFLAGS         =

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    makefile

build file:

    makefile

build file templates:

    Configurations\common0.tmpl
    Configurations\windows-makefile.tmpl
```

3. Refer openssl-3.0.5-build-error.txt file, to check the errors that I'm getting while executing command nmake
[openssl-3.0.5-build-error.txt](https://github.com/openssl/openssl/files/9886237/openssl-3.0.5-build-error.txt)

**Comment by levitte** [comment]
From the build output, it looks like you've set the prefix and openssldir to the source directory.  That's surprising.  Set them to a different directory, like `C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\install`

**Comment by rakesh-mahapatra** [comment]
I've tried that already (actually had that in my first try) but still encountering the same issue

**Comment by t8m** [comment]
Is this still relevant?

**Comment by njbelf** [comment]
I believe this is still an issue.   I am trying to set --prefix and --openssldir as arguments to the perl Configure command for a VC-WIN32 target.    The Configure seems to ignore the --prefix and --openssldir as I look at the generated makefile and it's always set to the default location.    I have tried quoting the path and both kinds of separators.

**Comment by nhorman** [comment]
but --prefix and --openssldir really shouldn't affect the build, and given that the error messages are (to cite a sample):

```
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\apps\include\app_libctx.h(15) : error C2143: syntax error : missing '{' before '*'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\apps\include\app_libctx.h(16) : error C2143: syntax error : missing '{' before '*'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\openssl/safestack.h(205) : error C2061: syntax error : identifier 'ossl_inline'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\openssl/safestack.h(205) : error C2059: syntax error : ';'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\openssl/safestack.h(205) : error C2061: syntax error : identifier 'ossl_inline'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\openssl/safestack.h(205) : error C2059: syntax error : ';'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\openssl/safestack.h(205) : error C2143: syntax error : missing '{' before '*'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\openssl/safestack.h(205) : error C2146: syntax error : missing ')' before identifier 'OPENSSL_STACK'
C:\VirtualPartition-D\Project\CLP\OpenSSL\OpenSSL_3.0\openssl-3.0.5-site-2022-Oct-12\include\o
```
it looks like some of your source tree has gotten corrupted somehow.  Can you post your app_libctx.h file please?

**Comment by nhorman** [comment]
no response, marking inactive, to be closed at the end of 3.4 dev barring further input
